src/js/layer.js

```diff
@@ -0,0 +1,67 @@
+const loginArea = document.querySelector('.header__login');
+const loginLayer = document.querySelector('.header-layer__popup--login');
+const loginExpandLayer = document.querySelector('.header-layer__popup--login-extend');
+const addressArea = document.querySelector('.header__shipping-address');
+const addressLayer = document.querySelector('.header-layer__popup--address');
+const dimBackground = document.querySelector('.dim');
+
+let isOpenAddresslayer = false;
+let isOpenLoginExpandLayer = false;
+
+function renderLoginLayer() {
+  document.addEventListener('DOMContentLoaded', () => {
+    setTimeout(() => {
+      loginLayer.classList.remove('hidden');
+    }, 1000);
+  });
+}
+
+function handleAddressMouseEnter() {
+  if (!isOpenAddresslayer) openLayer(addressLayer);
+  isOpenAddresslayer = true;
+}
+
+function handleLoginMouseEnter() {
+  loginLayer.classList.add('hidden');
+  if (!isOpenAddresslayer) openLayer(loginExpandLayer);
+  isOpenLoginExpandLayer = true;
+}
+
+function handleAddressMouseLeave(e) {
+  isOpenAddresslayer = false;
+  closeLayer(addressLayer, e);
+}
+
+function handleLoginMouseLeave(e) {
+  isOpenLoginExpandLayer = false;
+  closeLayer(loginExpandLayer, e);
+}
+
+function openLayer(modal) {
+  modal.classList.remove('hidden');
+  dimBackground.classList.remove('hidden');
+}
+
+function closeLayer(modal, e) {
+  if (e.relatedTarget !== modal && !isOpenAddresslayer && !isOpenLoginExpandLayer) {
+    modal.classList.add('hidden');
+    dimBackground.classList.add('hidden');
+  }
+}
```

Comment:
3가지 조건을 한꺼번에 + 부정문으로 생각하는 게 한눈에 이해하기 힘들 수 있어서
코드는 다소 길어지지만 early return 방식으로 작성해보셔도 좋을 것 같습니다. 🙂

```suggestion
function closeLayer(modal, e) {
  if (isOpenAddresslayer) {
    return;
  }
  if (isOpenLoginExpandLayer) {
    return;
  }
  if (e.relatedTarget === modal) {
    return;
  }
  
  modal.classList.add('hidden');
  dimBackground.classList.add('hidden');
}
```

조건문을 긍정형으로 작성한다면 하나로 묶어서 처리하는 것도 괜찮을 것 같네요.

```suggestion
function closeLayer(modal, e) {
  if (isOpenAddresslayer || isOpenLoginExpandLayer || e.relatedTarget === modal) {
    return;
  }
  
  modal.classList.add('hidden');
  dimBackground.classList.add('hidden');
}
```

Reply:
이 부분도 고민을 했었는데.. 이렇게 긍정형으로도 해결이 되는군요!!
반대로 해결해볼 생각을 못했습니다 감사합니다 👍